libraries/standard/mqtt/utest/mqtt_utest.c

```diff
@@ -31,6 +54,150 @@ int suiteTearDown( int numFailures )
     return numFailures;
 }
 
+/* ========================================================================== */
+
+/**
+ * @brief Mock successful transport send, and write data into buffer for
+ * verification.
```

Comment:
That's clever!

Reply:
Thanks, but can't really take credit for this since I got the idea from our old unit tests in [v4_beta](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/v4_beta/libraries/standard/mqtt/test/unit/iot_tests_mqtt_api.c#L540)